packages/alfa-dom/src/get-tab-sequence.ts

```diff
@@ -0,0 +1,46 @@
+import { getTabIndex } from "./get-tab-index";
+import { traverseNode } from "./traverse-node";
+import { Element, NodeType } from "./types";
+
+function binaryIndexSearch(array: Array<Element>, index: number) {
+  let low = 0;
+  let high = array.length;
+
+  if (index === 0) {
+    return high;
+  }
+
+  while (low < high) {
+    const mid = (low + high) >>> 1;
+    const other = <number>getTabIndex(array[mid]);
```

Comment:
To avoid clashes with TSX syntax, type assertions should be made using the `as` keyword:

```ts
getTabIndex(array[middle]) as number
```

I'd also add a comment stating why the assertion is safe.